src/test/java/com/exactpro/th2/mstore/TestCaseMessageStore.java

```diff
@@ -218,7 +229,70 @@ void normalDelivery() throws IOException {
         assertStoredMessageBatch(value, "test", Direction.FIRST, 2);
     }
 
+    @Test
+    @DisplayName("Close message store when feature is complited")
+    void complitedFutureComplited() throws InterruptedException, ExecutionException, TimeoutException {
+        M first = createMessage("test", Direction.FIRST, 1);
+
+        messageStore.handle(deliveryOf(first));
+        messageStore.dispose();
+
+        verify(completableFuture).get(any(long.class), any(TimeUnit.class));
+    }
+
+    @Test
+    @DisplayName("Close message store when feature throws TimeoutException")
+    void complitedFutureTimeoutException() throws InterruptedException, ExecutionException, TimeoutException {
+        when(completableFuture.get(any(long.class), any(TimeUnit.class))).thenThrow(mock(TimeoutException.class));
+        when(completableFuture.cancel(any(boolean.class))).thenAnswer((invocation) -> {
+            when(completableFuture.isDone()).thenReturn(true);
```

Comment:
You can specify multiple answers for a method. They will be used in the order you declare them.

```java
when(completableFuture.isDone()).thenReturn(false, true)
```

So, you can extract `completableFuture.isDone()` stubbing to the top level for that test